### PR TITLE
feat: AssociationState Protos for Validation Service

### PR DESCRIPTION
**Pull request by insipx**

### CODE REVIEW

**Review by neekolas — APPROVED** [review]

File: proto/mls_validation/v1/service.proto [hunk 1]
```diff
@@ -59,3 +64,24 @@ message ValidateGroupMessagesResponse {
...
+
+// Map of members belonging to an inbox_id
+message MemberMap {
+  xmtp.identity.associations.MemberIdentifier key = 1;
+  xmtp.identity.associations.Member value = 2;
+}
+
+// Reponse to GetAssociationStateRequest, containing the final association state
+// for an InboxID
+message GetAssociationStateResponse {
```
Comment (richardhuaaa) [inline proto/mls_validation/v1/service.proto:82]: The server will need to know which addresses were affected, and whether they were associations or revocations, for the last appended identity update. What would be the best way to expose this?

**Review by richardhuaaa — CHANGES_REQUESTED** [review]

File: proto/mls_validation/v1/service.proto (hunk 1 above)
Comment (richardhuaaa) [reply proto/mls_validation/v1/service.proto:82]: https://github.com/xmtp/xmtp-node-go/blob/main/pkg/identity/api/v1/service.go#L53-L56

Comment (richardhuaaa) [reply proto/mls_validation/v1/service.proto:82]: Perhaps an optional repeated field in `GetAssociationStateRequest` for `new_updates` or similar? Open to feedback here

Comment (richardhuaaa) [reply proto/mls_validation/v1/service.proto:82]: I think we would also need this on the client for commit construction and validation: https://community.xmtp.org/t/xip-46-multi-wallet-identity/639#validating-the-other-proposals-in-the-commit-31

Perhaps the `GetAssociationStateRequest` can just have an `old_sequence_id` field, with all changes after that being tracked?

**Comment by richardhuaaa** [comment]
Also, if validation fails, it will fail on a specific sequence number, and I assume you'll abort the rest of the process. Could we get the sequence number that failed, and the state + changes up to that point? This will help with handling concurrency on the server

**Review by neekolas — APPROVED** [review]

**Review by richardhuaaa — APPROVED** [review]

**Pull request merged**